### PR TITLE
Initial gas fix

### PR DESCRIPTION
**Pull request by apoorvsadana**
- fixes incorrect initial_gas for transactions which would cause transactions to go through in the blockifier even if enough gas was not present
- fixed a bug where fees wouldn't be charged if max_fee was 0

### CODE REVIEW

File: crates/primitives/starknet/src/transaction/mod.rs [hunk 1]
```diff
@@ -429,7 +429,7 @@ impl Transaction {
         };
 
         // FIXME 710
-        let mut initial_gas = super::constants::INITIAL_GAS;
+        let mut initial_gas = self.max_fee.try_into().unwrap(); // unwrap safe as >u64 case will be handled at client side
```
Comment (d-roak) [inline crates/primitives/starknet/src/transaction/mod.rs:432]: shouldn't it be independent of the client?

File: .vscode/launch.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
       // As such, this path would have to change accordingly.
       "program": "${workspaceFolder}/target/debug/madara",
       // If you wish to supply arguments/parameters to the program, supply them below:
-      "args": ["--dev", "--sealing=instant"],
+      "args": ["--dev", "--sealing=instant", "--tmp", "--execution=Native"],
```
Comment (d-roak) [inline .vscode/launch.json:21]: why are u using native execution?

File: crates/primitives/starknet/src/transaction/mod.rs [hunk 3]
```diff
@@ -555,8 +555,11 @@ impl Transaction {
         // Verify the transaction version.
         self.verify_tx_version(&tx_type)?;
 
-        // FIXME 710
-        let mut initial_gas = super::constants::INITIAL_GAS;
+        // if it's an estimate fee then use max initial_gas
+        let mut initial_gas = match self.is_query {
+            true => u64::MAX,
+            false => self.max_fee.try_into().unwrap(), // unwrap safe as >u64 case will be handled at client side
```
Comment (d-roak) [inline crates/primitives/starknet/src/transaction/mod.rs:561]: same as above

File: .vscode/launch.json (hunk 2 above)
Comment (apoorvsadana) [reply .vscode/launch.json:21]: this is for launch.json which is used when using the vs code debugger. the debugger won't work if the execution isn't native

File: crates/primitives/starknet/src/transaction/mod.rs (hunk 1 above)
Comment (apoorvsadana) [reply crates/primitives/starknet/src/transaction/mod.rs:432]: what should be independent of the client?

Comment (d-roak) [reply crates/primitives/starknet/src/transaction/mod.rs:432]: the safeness (or not) of the unwrap

File: CHANGELOG.md [hunk 4]
```diff
@@ -29,6 +29,7 @@
 - chore: cairo-contracts compilation scripts & docs are updated, cairo_0
   contracts recompiled
 - chore: rebase of core deps and 0.12.1
+- fix: initial_gas set to max_fee and fixed fee not being charged when max_fee=0
```
Comment (tdelabro) [inline CHANGELOG.md:32]: Wrong spot. It goes under `Next release`

**Review by tdelabro — APPROVED** [review]

File: crates/primitives/starknet/src/transaction/mod.rs (hunk 1 above)
Comment (apoorvsadana) [reply crates/primitives/starknet/src/transaction/mod.rs]: makes sense, fixed

File: crates/primitives/starknet/src/transaction/mod.rs (hunk 3 above)
Comment (apoorvsadana) [reply crates/primitives/starknet/src/transaction/mod.rs]: fixed

File: CHANGELOG.md (hunk 4 above)
Comment (apoorvsadana) [reply CHANGELOG.md]: fixed

**Pull request merged**